### PR TITLE
Webumenia 1925 connect to be

### PR DESCRIPTION
**Pull request by FrantisekMichalSebestyen**
Just a draft. I would like to ask about few stuff

### CODE REVIEW

File: resources/js/components/catalog-new/ItemsFilterController.vue [hunk 1]
```diff
@@ -1,43 +1,49 @@
 <script>
-import queryString from 'query-string'
-//TODO: import axios from 'axios'
+import qs from 'qs'
```
Comment (FrantisekMichalSebestyen) [inline resources/js/components/catalog-new/ItemsFilterController.vue:2]: Switch to qs as queryString does not support objects in url, which are needed for api

File: resources/js/components/catalog-new/ItemsFilterController.vue [hunk 2]
```diff
@@ -1,43 +1,49 @@
...
 
 function stringifyUrl({ url, query }) {
-    return queryString.stringifyUrl(
-        { url, query },
-        {
-            arrayFormat: 'bracket',
-            skipNull: true,
-        }
-    )
+    const newQuery = {
```
Comment (FrantisekMichalSebestyen) [inline resources/js/components/catalog-new/ItemsFilterController.vue:13]: WDYT about this? search part of the url is not the same as we have in Controller. However keeping date_earliest in object would complicate stuff quite a lot in different part of code

File: resources/js/components/catalog-new/ItemsFilterController.vue [hunk 3]
```diff
@@ -1,43 +1,49 @@
...
+    return url + '?' + qs.stringify(newQuery, { skipNulls: true, arrayFormat: 'brackets' })
 }
 
-const singleItemFilters = ['color', 'yearFrom', 'yearTo']
-const emptyQuery = {
-    authors: [],
+const SIZE = 50
+const SINGLE_ITEM_FILTERS = ['color', 'yearFrom', 'yearTo']
+
+const EMPTY_QUERY = {
```
Comment (FrantisekMichalSebestyen) [inline resources/js/components/catalog-new/ItemsFilterController.vue:28]: WDYT about constanst in uppercase

File: resources/js/components/catalog-new/ItemsFilterController.vue [hunk 4]
```diff
@@ -1,43 +1,49 @@
...
+        terms: query.terms,
+        size: query.size,
+    }
+    return url + '?' + qs.stringify(newQuery, { skipNulls: true, arrayFormat: 'brackets' })
 }
 
-const singleItemFilters = ['color', 'yearFrom', 'yearTo']
-const emptyQuery = {
-    authors: [],
+const SIZE = 50
```
Comment (FrantisekMichalSebestyen) [inline resources/js/components/catalog-new/ItemsFilterController.vue:25]: Are we gonna paginate aggregations ?

File: resources/js/components/catalog-new/ItemsFilterController.vue [hunk 5]
```diff
@@ -134,19 +162,62 @@ export default {
...
-                // TODO: Fetch options
-                // TODO: Fetch artworks
+
+                this.artworks = await axios
+                    .get(
+                        stringifyUrl({
+                            url: '/api/v1/items',
+                            params: {
+                                filter: this.query,
+                                size: SIZE_ARTWORK,
```
Comment (eronisko) [inline resources/js/components/catalog-new/ItemsFilterController.vue:216]: Does this need the `page` parameter too? (It's OKay if that's part of a future task)

Comment (FrantisekMichalSebestyen) [reply resources/js/components/catalog-new/ItemsFilterController.vue:216]: yes, left it for pagination

File: resources/js/components/catalog-new/ItemsFilterController.vue [hunk 6]
```diff
@@ -1,43 +1,74 @@
...
+        ...yearRange,
+    }
+    return url + '?' + qs.stringify(newQuery, { skipNulls: true, arrayFormat: 'brackets' })
 }
 
-const singleItemFilters = ['color', 'yearFrom', 'yearTo']
-const emptyQuery = {
-    authors: [],
+const PAGE_SIZE = 30
+const AGGREGATION_SIZE = 1000
```
Comment (eronisko) [inline resources/js/components/catalog-new/ItemsFilterController.vue:39]:
```suggestion
const AGGREGATIONS_SIZE = 1000
```

**Review by eronisko — APPROVED** [review]

**Pull request merged**